core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/parser/antlr/DataRepoInputExpressionValidationVisitor.scala

```diff
@@ -19,45 +22,72 @@ class DataRepoInputExpressionValidationVisitor(rootEntityType: Option[String],
+      ))
     }
   }
 
-  private def checkForAttributeOnTable(tableModel: TableModel, attributeNameContext: AttributeNameContext): Try[Unit] = {
+  private def checkForAttributeOnTable(tableModel: TableModel, bigQueryAttributeNameContext: BigQueryAttributeNameContext): Try[Unit] = {
     val tableColumns = tableModel.getColumns.asScala.toList
-    val attributeName = attributeNameContext.getText
+    val attributeName = bigQueryAttributeNameContext.getText
     if (tableColumns.exists(_.getName == attributeName)) {
       Success()
     } else {
-      Failure(new RawlsException(s"Missing attribute `${attributeName}` on table `${tableModel.getName}`"))
+      Failure(new RawlsExceptionWithErrorReport(
+        ErrorReport(StatusCodes.BadRequest, s"Missing attribute `${attributeName}` on table `${tableModel.getName}`")
+      ))
     }
   }
 
-  // TODO: CA-939 support relationships once that information is published by data repo
-  private def traverseRelationsAndGetFinalTable(currentTableModel: TableModel, relations: List[RelationContext]): Try[TableModel] = {
+  @tailrec
+  private def traverseRelationsAndGetFinalTable(currentTableModel: TableModel, relations: List[BigQueryRelationContext]): Try[TableModel] = {
     relations match {
       case Nil => Success(currentTableModel)
-      case _ => Failure(new RawlsException(s"Relationships are not currently supported."))
+      case nextRelationContext :: remainingRelations => {
+        val nextRelationName = nextRelationContext.bigQueryAttributeName().name.getText
+        maybeGetNextTableFromRelation(currentTableModel, nextRelationName) match {
+          case Some(nextTableModel) => traverseRelationsAndGetFinalTable(nextTableModel, remainingRelations)
+          case None => Failure(new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.BadRequest, s"Relationship with name `${nextRelationName}` and from table `${currentTableModel}` could not be found.") // This would only happen if there is a bug in TDR code.
+          ))
+        }
+      }
     }
   }
 
   private def maybeFindTableInSnapshotModel(tableName: String): Option[TableModel] = {
     val snapshotTables = snapshotModel.getTables.asScala.toList
     snapshotTables.find(_.getName == tableName)
   }
+
+  private def maybeGetNextTableFromRelation(fromTable: TableModel, relationName: String): Option[TableModel] = {
+    val relationships = snapshotModel.getRelationships.asScala.toList
+    maybeFindTableInSnapshotModel(fromTable.getName)
+      .flatMap { tableModel =>
+        relationships.find { relationship =>
+          relationship.getFrom.getTable == tableModel.getName && relationship.getFrom.getColumn == relationName
+        }.map { relationship =>
+          val nextTableName = relationship.getTo.getTable
+          return maybeFindTableInSnapshotModel(nextTableName)
+        }
+      }
+  }
```

Comment:
3 things here:
1) try rewriting this as a for comprehension. More than one flapMap/map works better as a for comp
2) this needs to match the relationship in both directions. Either `relationship.getFrom.getTable == tableModel.getName` or `relationship.getTo.getTable == tableModel.getName`
3) `relationship.getFrom.getColumn == relationName` is not right, the relation name _is not_ the relation name